src/components/Task.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <li
+    class="
+      relative
+      bg-white
+      py-5
+      px-4
+      transition
+      duration-300
+      ease-in-out
+      flex
+      justify-between
+      items-center
+    "
+  >
+    <input
+      :id="currentTask.id"
+      v-model="currentTask.completed"
+      type="checkbox"
+      class="
+        rounded
+        text-blue-500
+        mr-10
+        pointer-events-auto
+        focus:outline-none
+        focus:ring-transparent
+        hover:cursor-pointer
+        transition
+        duration-300
+        ease-in-out
+      "
+    />
+    <div class="flex-grow">
+      <div class="flex justify-between items-stretch space-x-3">
+        <div class="flex-1">
+          <div class="min-w-0 flex-1">
+            <div
+              class="block focus:outline-none cursor-pointer"
+              :class="{
+                'pointer-events-none resize-none': currentTask.completed,
+              }"
+              @click="$emit('open:detail', currentTask.id)"
+            >
+              <p
+                class="text-lg font-medium text-gray-900 truncate"
+                :class="{ 'text-gray-600 line-through': currentTask.completed }"
```

Comment:
@tupic98 good one, I like this. 

But it would be way better if you created a `computed` value and pass it here instead of using plain object, with that, all the logic stay in JS part and template part would be cleaner.